### DISCUSSION TITLE
[@types/react] - Compile errors on TypeScript 3.8.x

### DISCUSSION THREAD

**Discussion by robrez** [DefinitelyTyped/DefinitelyTyped#69207] [Issues with a @types package] [answered]
More of an FYI, feel free to close immediately. Mostly here for others who may have the same issue -

I have a repo w/ some technical debt that still uses typescript `@^3.x.x`...  since [b121e7c](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/b121e7ceeaaf415fdf15146273fbe8361d3c6b4d), this repo (which has a transitive dependency on `@types/react` no longer compiles

```bash
../../../node_modules/@types/react/ts5.0/index.d.ts(3229,50): error TS1110: Type expected.
../../../node_modules/@types/react/ts5.0/index.d.ts(3230,51): error TS1110: Type expected.
../../../node_modules/@types/react/ts5.0/index.d.ts(3231,48): error TS1110: Type expected.
../../../node_modules/@types/react/ts5.0/index.d.ts(3231,91): error TS1005: '(' expected.
../../../node_modules/@types/react/ts5.0/index.d.ts(3232,28): error TS1110: Type expected.
../../../node_modules/@types/react/ts5.0/index.d.ts(3235,11): error TS1110: Type expected.
../../../node_modules/@types/react/ts5.0/index.d.ts(3235,50): error TS1005: '(' expected.
../../../node_modules/@types/react/ts5.0/index.d.ts(3237,10): error TS1005: '(' expected.
../../../node_modules/@types/react/ts5.0/index.d.ts(3237,74): error TS1005: '(' expected.
```

Remedial action

```json
// package.json
{
  "devDependencies": {
     "@types/react": "^18.0.0 <=18.2.56",
     "@types/scheduler": "^0.16.8",
  }
}
```

Note that `@types/scheduler` was required but not present for some amount of time [see PR#69178](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69178/files)

_Edit_ On my particular repo (the one w/ the tech debt I mentioned) - I determined (based on its dependents) - that I could upgrade to `typescript@~4.8.0` without causing problems elsewhere. Doing so also remedies the problem noted above

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
TypeScript 3.8 is not supported by any `@types/*` package. The lowest, currently supported TypeScript version is 4.7. See https://github.com/definitelyTyped/DefinitelyTyped/#support-window for more information.

**Reply by robrez** [reply]
Thanks, Sebastian. That's another thing I can point to:  "You really need to upgrade your dep graph".
